src/main/java/gregtech/common/gui/widget/CompileStatusWidget.java

```diff
@@ -0,0 +1,82 @@
+package gregtech.common.gui.widget;
+
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.gui.IRenderContext;
+import gregtech.api.gui.Widget;
+import gregtech.api.gui.resources.TextureArea;
+import gregtech.api.gui.widgets.TextFieldWidget2;
+import gregtech.api.util.oreglob.OreGlobCompileResult;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.TextFormatting;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CompileStatusWidget extends Widget {
```

Comment:
This is named universally which can be misleading as it is using ore dictionary specific lang keys and textures.

I think it should be considered if this could be done totally universally so it could be reused in future.

Reply:
It's probably better to change the classname to `OreExpressionCompileStatusWidget` or `OreGlobCompileStatusWidget` yeah. I doubt there will be another need for 'compilation's in the mod, so opening up the implementation seems redundant for now.